actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -1168,13 +1168,27 @@ module Resources
         CANONICAL_ACTIONS = %w(index create new show update destroy)
 
         class Resource # :nodoc:
+          class << self
+            def default_actions(api_only)
+              if api_only
+                [:index, :create, :show, :update, :destroy]
+              else
+                [:index, :create, :new, :show, :update, :destroy, :edit]
+              end
+            end
+          end
+
```

Comment:
Not tested:
```suggestion
        API_ONLY_DEFAULT_ACTIONS = [:index, :create, :show, :update, :destroy].freeze
        DEFAULT_ACTIONS = (API_ONLY_DEFAULT_ACTIONS + [:new, :edit]).freeze
        private_constant :API_ONLY_DEFAULT_ACTIONS, :DEFAULT_ACTIONS

        class Resource # :nodoc:
          class << self
            def default_actions(api_only)
              return API_ONLY_DEFAULT_ACTIONS if api_only

              DEFAULT_ACTIONS
            end
          end

```

Reply:
Hmm personally I think the current version is much easier to follow. I'd also like to keep cosmetic changes/refactors to a minimum to allow for an easier review.